proxy/src/main/java/com/velocitypowered/proxy/connection/client/ClientConfigSessionHandler.java

```diff
@@ -105,6 +108,31 @@ public boolean handle(FinishedUpdate packet) {
     return true;
   }
 
+  @Override
+  public boolean handle(PluginMessage packet) {
+    VelocityServerConnection serverConn = player.getConnectionInFlight();
+    if (serverConn != null) {
+      if (PluginMessageUtil.isMcBrand(packet)) {
+        String brand = PluginMessageUtil.readBrandMessage(packet.content());
+        server.getEventManager().fireAndForget(new PlayerClientBrandEvent(player, brand));
+        player.setClientBrand(brand);
+        // Client sends `minecraft:brand` packet immediately after Login,
```

Comment:
We should probably look into ensuring that this is at least sent somewhere if we can't just send it right now; I need to spend some time looking over this, ideally we should race hard to get the connection to the server into configuration so we can deal with syncing stuff, but, that is probably a wider headache to deal with someday (hopefully soon™)